pushjack/apns.py

```diff
@@ -490,16 +503,33 @@ def to_dict(self):
 
             alert = compact_dict(alert)
         else:
-            alert = self.message
+            alert = {}
 
         message.update(self.extra or {})
-        message['aps'] = compact_dict({
-            'alert': alert,
-            'badge': self.badge,
-            'sound': self.sound,
-            'category': self.category,
-            'content-available': 1 if self.content_available else None
-        })
+
+        if alert:
```

Comment:
I'm not sure this `if/else` block is necessary. The way that `compact_dict()` works is that it will remove any top-level keys that are set to `None`. So if `message['alert'] == None`, then the `'alert'` key would be removed by `compact_dict()`.